packages/overlay/src/OverlayBase.ts

```diff
@@ -0,0 +1,908 @@
+        flatten: true,
+        selector: ':not([slot="longpress-describedby-descriptor"])', // gather only elements slotted into the default slot
+    })
+    elements!: OpenableElement[];
+
+    public parentOverlayToForceClose?: OverlayBase;
+
+    private get hasNonVirtualTrigger(): boolean {
+        return (
+            !!this.triggerElement &&
+            !(this.triggerElement instanceof VirtualTrigger)
+        );
+    }
+
+    protected longpressed = false;
+
+    private longressTimeout!: ReturnType<typeof setTimeout>;
+
+    @property({ type: Number })
+    offset: number | [number, number] = 6;
+
+    public placementController = new PlacementController(this);
+
+    @property({ type: Boolean, reflect: true })
+    get open(): boolean {
+        return this._open;
+    }
+
+    set open(open: boolean) {
+        if (open && this.disabled) return;
+        if (open === this.open) return;
+        this._open = open;
+        if (this.open) {
+            OverlayBase.openCount += 1;
+        }
+        this.requestUpdate('open', !this.open);
+    }
+
+    private _open = false;
+
+    static openCount = 1;
+
+    @property()
+    placement?: Placement;
+
+    @property({ attribute: 'receives-focus' })
+    receivesFocus: 'true' | 'false' | 'auto' = 'auto';
+
+    private releaseAriaDescribedby = noop;
+    private releaseLongpressDescribedby = noop;
+
+    @query('slot')
+    slotEl!: HTMLSlotElement;
+
+    @property({ type: Number, attribute: 'tip-padding' })
+    tipPadding?: number;
+
+    @property()
+    trigger?: string;
+
+    @state()
+    triggerElement: HTMLElement | VirtualTrigger | null = null;
+
+    @state()
+    triggerInteraction?: 'click' | 'longpress' | 'hover';
+
+    @property()
+    type: OverlayTypes = 'hint';
+
+    protected wasOpen = false;
+
+    private elementResolver = new ElementResolutionController(this);
+
+    private get usesDialog(): boolean {
+        return this.type === 'modal' || this.type === 'page';
+    }
+
+    private get popoverValue(): 'auto' | 'manual' | undefined {
+        switch (this.type) {
+            case 'modal':
+            case 'page':
+                return undefined;
+            case 'hint':
+                return 'manual';
+            default:
+                return this.type;
+        }
+    }
+
+    /* c8 ignore next 12 */
+    protected async manageDialogOpen(): Promise<void> {
+        console.warn(
+            'Implement the `manageDialogOpen` method in a class extension.'
+        );
+    }
+
+    protected async managePopoverOpen(): Promise<void> {
+        console.warn(
+            'Implement the `managePopoverOpen` method in a class extension.'
+        );
+    }
+
+    protected get requiresPosition(): boolean {
+        // Do not position "page" overlays as they should block the entire UI.
+        if (this.type === 'page' || !this.open) return false;
+        // Do not position content without a trigger element, what would you position it in relation to?
```

Comment:
Seems like this is one of the things to document for https://github.com/adobe/spectrum-web-components/pull/3456#discussion_r1268611639